lib/rules/require-matching-function-name.js

```diff
@@ -96,11 +96,13 @@ module.exports.prototype = {
 };
 
 function checkForMember(assignment, skip, errors) {
-    if (skip(assignment.left.property.name, assignment.right)) {
+    var _left = assignment.left;
+    var _name = _left.property ? (_left.property.name || _left.property.value) : _left.name;
```

Comment:
Non-blocker: For readability, I find it helpful for maintenance when an example code is supplied that clarifies an assignment with a left property that doesn't have a name but has a value. Similarly, what's an assignment that doesn't have a property?

A contributor would have to guess or hope that coverage is 100% and figure out which tests correspond to which part of the condition.